mypy/plugins/attrs.py

```diff
@@ -192,6 +193,21 @@ def attr_class_maker_callback(ctx: 'mypy.plugin.ClassDefContext',
 
     attributes = _analyze_class(ctx, auto_attribs)
 
+    # Trigger a "wildcard" update for the class whenever one of
+    # its attributes changes.  In collect_attributes, we arranged
+    # for each class (or rather the defining module) to depend on
+    # the wildcard trigger of every dataclass that it inherits
+    # from.  This will cause all subclasses to get reprocessed
+    # whenever a dataclass attribute in a parent is added,
+    # removed, or changed.
+    #
+    # This is somewhat annoyingly subtle. An alternate approach would be
+    # to have a dependency from each attribute to a full class trigger,
+    # which is a little simpler but will force rechecks in too much code.
+    for attr in attributes:
+        ctx.api.add_plugin_dependency(make_trigger(info.fullname() + '.' + attr.name),
+                                      make_wildcard_trigger(info.fullname()))
```

Comment:
This looks quite weird. The whole point of a wildcard trigger is so it is triggered if anything changes. This is very similar to what is needed for protocols and in my case it was sufficient to just add `<Super[wildcard]> -> <Sub>`.

If it is not triggered maybe it is a bug in `calculate_active_triggers` (I remember I added some wildcard code there for protocols, before it only worked for modules).

Reply:
Yeah, this is all nonsense. I thought that `[wildcard]` only worked for modules, and I didn't want to modify `calculate_active_triggers` since I wanted to demonstrate it could all be done as a plugin.